app/controllers/api/predictions_controller.rb

```diff
@@ -52,8 +70,28 @@ def build_predictions
       end
     end
 
-    def invalid_message
+    def find_prediction
+      @prediction = Prediction.find(params[:id])
+    end
+
+    def invalid_api_message
       { error: 'invalid API token', status: :unauthorized }
     end
+
+    def unauthorized_user_message
+      {
+        error: 'user is unauthorized to view this private prediction',
+        status: :unauthorized
+      }
+    end
+
+    def successfully_updated?
+      return false unless @user.authorized_for(@prediction)
```

Comment:
Rather than adding the auth check here, I would prefer it added to the top of the methods which are using it. It's the first thing that's checked before the action is processed.